File: app/src/main/java/com/psdk/DisplayElfInfo.java
package com.psdk;

import java.io.File;
import java.util.HashMap;

public class DisplayElfInfo {

	private final static String ARMV7ABI = "armeabi-v7a";
	private final static String X86 = "x86";
	private final static String MIPS = "mips";
	private final static String X86_64 = "x86_64";
	private final static String ARM64_V8 = "arm64-v8a";
	private final static String ARMABI = "armeabi";

	private static HashMap<Integer, String> typeMap = new HashMap<>();

	public static String findAbiType(File libFile) {
		if (!libFile.exists()) {
			return "";
		}

		if (typeMap.isEmpty()) {
			initializeMap();
		}

		try {
			ReadElf elf = ReadElf.read(libFile);
			if (elf.getType() == 3) {
				int archCode = elf.getArch();
				elf.close();
				return typeMap.get(archCode);
			}
			elf.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
		return "";
	}

	private static void initializeMap() {
		typeMap.put(40, ARMV7ABI);
		typeMap.put(3, X86);
		typeMap.put(8, MIPS);
		typeMap.put(62, X86_64);
		typeMap.put(183, ARM64_V8);
		typeMap.put(164, ARMABI);
	}
}

File: app/src/main/java/com/psdk/DisplayElfInfo.kt
package com.psdk

import java.io.File
import java.lang.Exception
import java.util.HashMap

object DisplayElfInfo {
    private const val ARMV7ABI = "armeabi-v7a"
    private const val X86 = "x86"
    private const val MIPS = "mips"
    private const val X86_64 = "x86_64"
    private const val ARM64_V8 = "arm64-v8a"
    private const val ARMABI = "armeabi"
    private val typeMap = HashMap<Int, String>()
    fun findAbiType(libFile: File): String? {
        if (!libFile.exists()) {
            return ""
        }
        if (typeMap.isEmpty()) {
            initializeMap()
        }
        try {
            val elf: ReadElf = ReadElf.Companion.read(libFile)
            if (elf.type == 3) {
                val archCode = elf.arch
                elf.close()
                return typeMap[archCode]
            }
            elf.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return ""
    }

    private fun initializeMap() {
        typeMap[40] = ARMV7ABI
        typeMap[3] = X86
        typeMap[8] = MIPS
        typeMap[62] = X86_64
        typeMap[183] = ARM64_V8
        typeMap[164] = ARMABI
    }
}